django/template/defaulttags.py

```diff
@@ -1220,6 +1239,34 @@ def regroup(parser, token):
 
 
 @register.tag
+def resetcycle(parser, token):
+    """
+    Resets a cycle tag.
+
+    If an argument is given, resets the last rendered cycle tag whose name
+    matches the argument, else resets the last rendered cycle tag (named or
+    unnamed).
+    """
+    args = token.split_contents()
+
+    if len(args) > 2:
+        raise TemplateSyntaxError("%r tag accepts at most one argument." % args[0])
+
+    if len(args) == 2:
+        name = args[1]
+        try:
+            return ResetCycleNode(parser._named_cycle_nodes[name])
+        except AttributeError:
+            raise TemplateSyntaxError("No named cycles in template. %r is not defined" % name)
```

Comment:
I *think* that the intended important bit of the message here is that there are no named cycles at all in the template, and thus the named cycle is not defined. I'm happy to change it, as it is a bit more readable, as long as it doesn't distract fromt the other error messages where there are named cycles available, but not this one.

@timgraham your thoughts?